tests/modes/test_abstract_mode_consumer.py

```diff
@@ -169,6 +173,70 @@ async def test_get_holdings_ratio():
+    exchange_manager.exchange_personal_data.portfolio_manager.portfolio.portfolio["USDT"] = \
+        portfolio_assets.SpotAsset(name="USDT", available=decimal.Decimal("1000"), total=decimal.Decimal("1000"))
+
+    assert consumer.get_holdings_ratio("BTC", include_assets_in_open_orders=False) \
+           == decimal.Decimal('0.9090909090909090909090909091')
+    assert consumer.get_holdings_ratio("USDT", include_assets_in_open_orders=False) \
+           == decimal.Decimal('0.09090909090909090909090909091')
+
+    # no open order
+    assert consumer.get_holdings_ratio("BTC", include_assets_in_open_orders=True) \
+           == decimal.Decimal('0.9090909090909090909090909091')
+    assert consumer.get_holdings_ratio("USDT", include_assets_in_open_orders=True) \
+           == decimal.Decimal('0.09090909090909090909090909091')
+
+    order_1 = mock.Mock(
+        order_id="1", symbol="BTC/USDT", origin_quantity=decimal.Decimal("1"), total_cost=decimal.Decimal('1000'),
+        status=enums.OrderStatus.OPEN, side=enums.TradeOrderSide.BUY
+    )
+    order_2 = mock.Mock(
+        order_id="2", symbol="ETH/USDT", origin_quantity=decimal.Decimal("2"), total_cost=decimal.Decimal('100'),
+        status=enums.OrderStatus.OPEN, side=enums.TradeOrderSide.BUY
+    )
+    order_3 = mock.Mock(
+        order_id="3", symbol="XRP/ETH", origin_quantity=decimal.Decimal("100"), total_cost=decimal.Decimal('0.001'),
+        status=enums.OrderStatus.OPEN, side=enums.TradeOrderSide.SELL
+    )
+    # open orders
+    await exchange_manager.exchange_personal_data.orders_manager.upsert_order_instance(order_1)
+    await exchange_manager.exchange_personal_data.orders_manager.upsert_order_instance(order_2)
+    assert consumer.get_holdings_ratio("BTC", include_assets_in_open_orders=False) \
+           == decimal.Decimal('0.9090909090909090909090909091')
+    assert consumer.get_holdings_ratio("ETH", include_assets_in_open_orders=False) \
+           == decimal.Decimal('0')
+    assert consumer.get_holdings_ratio("USDT", include_assets_in_open_orders=False) \
+           == decimal.Decimal('0.09090909090909090909090909091')
+
+    assert consumer.get_holdings_ratio("BTC", include_assets_in_open_orders=True) \
+           == decimal.Decimal('1')
+    assert consumer.get_holdings_ratio("ETH", include_assets_in_open_orders=True) \
+           == decimal.Decimal('0.01818181818181818181818181818')
+    assert consumer.get_holdings_ratio("USDT", include_assets_in_open_orders=True) \
+           == decimal.Decimal('0.09090909090909090909090909091')
+
+    await exchange_manager.exchange_personal_data.orders_manager.upsert_order_instance(order_3)
+    assert consumer.get_holdings_ratio("BTC", include_assets_in_open_orders=True) \
+           == decimal.Decimal('1')
+    assert consumer.get_holdings_ratio("ETH", include_assets_in_open_orders=True) \
+           == decimal.Decimal('0.01819090909090909090909090909')
```

Comment:
I don't understand why this value doesn't change due to order_3 addition

Reply:
we are selling XRP for ETH in order3 making more ETH available when it will be filled

Reply:
So why the ETH ratio is the same as without order_3 ?

Reply:
it's not the same, before it was `0.01818181818181818181818181818` and after it's `0.01819090909090909090909090909`

Reply:
Oooh right, my bad. It saw the same number :smile: